src/components/util/overlays/popover/usePopover.ts

```diff
@@ -79,6 +79,35 @@ export const usePopover = <E extends HTMLElement>(
     }
   }, [controller.activate, controller.deactivate]);
   
+  // Added for outside click detection
+  React.useEffect(() => {
+  if (!controller.active) return;
+
+  const handleClickOutside = (event: MouseEvent) => {
+    const target = event.target as Node;
```

Comment:
Instance of the `as` (which may be unsafe, you can check if `event.target` is actually a node through `event.target instanceof Node`